File: Models/ComponentModel.cs
using System.Text.Json.Serialization;

namespace Gridly.Models;

public class ComponentModel
{
    [JsonPropertyName("id")] public int Id { get; set; }
    [JsonPropertyName("indexPosition")] public int IndexPosition { get; set; }
    [JsonPropertyName("name")] public string Name { get; set; }
    [JsonPropertyName("url")] public string Url { get; set; }
    [JsonPropertyName("iconData")] public IconModel? IconData { get; set; }
    [JsonPropertyName("iconUrl")] public string IconUrl { get; set; }
    [JsonPropertyName("componentSettings")] public ComponentSettingsModel? ComponentSettings { get; set; }
}

File: Gridly-Client/src/app/Models/Component.Model.ts
import {ComponentSettingsModel} from "./ComponentSettings.Model";
import {IconModel} from "./Icon.Model";

export class ComponentModel {
  id!: number;
  indexPosition!: number;
  name!: string;
  url!: string;
  iconData?: IconModel |
    {
      type: "",
      name: "",
      base64Data: "",
    };
  iconUrl?: string;
  componentSettings? : ComponentSettingsModel |
    {
      width:250,
      height:250,
      imageHidden:false,
      titleHidden:false
    };
}
